File: python/src/patterns/observer/observers.py
from .order_subject import OrderStatus

class EmailObserver:
    def __init__(self) -> None:
        self.messages: list[OrderStatus] = []
    def update(self, value: OrderStatus) -> None:
        self.messages.append(value)

class LogObserver:
    def __init__(self) -> None:
        self.events: list[OrderStatus] = []
    def update(self, value: OrderStatus) -> None:
        self.events.append(value)

File: typescript/src/patterns/observer/Observers.ts
import type { Observer } from './Observer.js';
import type { OrderStatus } from './OrderSubject.js';

export class EmailObserver implements Observer<OrderStatus> {
  public messages: OrderStatus[] = [];
  update(value: OrderStatus): void { this.messages.push(value); }
}

export class LogObserver implements Observer<OrderStatus> {
  public events: OrderStatus[] = [];
  update(value: OrderStatus): void { this.events.push(value); }
}
